src/tranche-space.rain

```diff
@@ -129,19 +129,29 @@ scenarios:
           # 1e16 is 1%
           tranche-space-snap-threshold: 1e16
 
-          # This is only relevant if the tranche size/ratio is denominated in
-          # some token other than the input/output tokens. For example, if the
-          # TKN was being traded for WETH but the tranche size was denominated in
-          # USD then the reference-stable would be USD and the reference-reserve
-          # would be WETH, and the identity multiplier needs to be swapped out
-          # for e.g. a TWAP USDT based multiplier.
-          # Typically this is NOT needed as the tranche size and ratio ARE
-          # denominated in the input/output tokens.
-          io-ratio-multiplier: '''io-ratio-multiplier-identity'
-
         scenarios:
           buy:
             bindings:
+                # This is only relevant if the tranche size/ratio is denominated in
+                # some token other than the input/output tokens. For example, if the
+                # TKN was being traded for WETH but the tranche size was denominated in
+                # USD then the reference-stable would be USD and the reference-reserve
+                # would be WETH, and the identity multiplier needs to be swapped out
+                # for e.g. a TWAP USDT based multiplier.
+                # Typically this is NOT needed as the tranche size and ratio ARE
+                # denominated in the input/output tokens.
+                io-ratio-multiplier: '''io-ratio-multiplier-identity'
+
+                # If TKN was being traded for some other token, for e.g WETH and if
+                # the tranche size/ratio was denominated in USD, then the reference and
+                # reserve bindings need to be given along with the twap bindings
+                reference-stable: 0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9
+                reference-stable-decimals: 6
```

Comment:
are the decimals needed?

Reply:
yes, they are passed as an argument for the uniswap twap call.